front/app/utils/streams.ts

```diff
@@ -429,19 +428,21 @@ class Streams {
     streamId: string,
     isQueryStream: boolean
   ) {
-    if (isQueryStream) {
-      if (!this.streamIdsByApiEndPointWithQuery[apiEndpoint]) {
-        this.streamIdsByApiEndPointWithQuery[apiEndpoint] = [streamId];
-      } else {
-        this.streamIdsByApiEndPointWithQuery[apiEndpoint].push(streamId);
-      }
```

Comment:
We didn't check  if the streamId was already in this array. Not sure if this is causing bugs.